web/src/routes/~_dashboard/DashboardLayout.tsx

```diff
@@ -51,6 +85,46 @@ export const DashboardLayout = ({
         mobile={sideMenu}
         {...props}
       />
+      <ConsentDialog open={!hasConsent}>
+        <DialogHeader>
+          <DialogTitle>Legal Disclaimer</DialogTitle>
+          <DialogDescription>
+            Please read and accept the terms before proceeding
+          </DialogDescription>
+        </DialogHeader>
+        <ConsentDialogContent>
+          <div className="space-y-2">
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sed
+              tortor nisl. Vestibulum a lectus quis libero condimentum maximus
+              ac sed sem. Cras vel ligula ac lorem commodo luctus. Phasellus at
+              convallis odio. Duis rhoncus felis vitae nisl pulvinar, quis
+              sagittis velit condimentum. Nullam lectus nulla, faucibus vitae
+              gravida a, venenatis non est. Nullam gravida, quam rhoncus dapibus
+              gravida, erat felis hendrerit arcu, posuere mattis sapien lectus
+              sit amet lectus. Nam faucibus velit ut ornare venenatis. Nunc
+              sagittis eleifend gravida. Cras in dignissim leo. Suspendisse
+              mauris orci, mattis quis eros finibus, cursus rhoncus velit.
+            </p>
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam sed
+              tortor nisl. Vestibulum a lectus quis libero condimentum maximus
+              ac sed sem. Cras vel ligula ac lorem commodo luctus. Phasellus at
+              convallis odio. Duis rhoncus felis vitae nisl pulvinar, quis
+              sagittis velit condimentum. Nullam lectus nulla, faucibus vitae
+              gravida a, venenatis non est. Nullam gravida, quam rhoncus dapibus
+              gravida, erat felis hendrerit arcu, posuere mattis sapien lectus
+              sit amet lectus. Nam faucibus velit ut ornare venenatis. Nunc
+              sagittis eleifend gravida. Cras in dignissim leo. Suspendisse
+              mauris orci, mattis quis eros finibus, cursus rhoncus velit.
+            </p>
+          </div>
+        </ConsentDialogContent>
+        <ConsentDialogCheckbox label="I have read and agree to the privacy policy" />{" "}
```

Comment:
This shouldn't be necessary here

```suggestion
        <ConsentDialogCheckbox label="I have read and agree to the privacy policy" />
```